### DISCUSSION TITLE
A mpv lua script to generate subtitles at runtime with whisper.cpp main

### DISCUSSION THREAD

**Discussion by GhostNaN** [ggml-org/whisper.cpp#843] [Show and tell]
Do you like watching videos with mpv, but don't like waiting around while whisper.cpp generates subtitles for you?
Well, I made WhisperSubs to fix this very niche issue. 

Check it out here: https://github.com/GhostNaN/whisper-subs

I originally planned on making a whole C++ program to make this process faster.
But as it turns out, you can speed up the processing time by making the wav file smaller and processing in chunks instead.
So, I just used the provided main example that comes with whisper.cpp and created this instead.

My PC is good enough to run the medium model fast enough to keep up, but your PC might vary.
Also please read the README on GitHub as I also explain the other static variables you will want to change.

In any case, I hope this weird lua script is at least interesting to some of you.

**Comment by dgoryeo** [comment]
Hi @GhostNaN , the readme suggests that variables TMP_WAV_PATH, TMP_SUB_PATH, TMP_STREAM_PATH can be left as they are. Am I right to say that for Windows environment they would definitely need to be changed?

**Reply by GhostNaN** [reply]
There would have to be many changes besides those variables to get this to work on Windows, but it isn't impossible.

At a minimum, it would also require modifying all the ```os.execute()``` and  ```io.popen()``` to work in a Windows environment.
Then yes, TMP_WAV_PATH, TMP_SUB_PATH, TMP_STREAM_PATH would all need to point to a temp folder in Windows.
The only thing left that would be an issue is my implementation of ```checkStreamStatus()``` function and streaming in general. 
I have no idea how that could work on Windows.

Since I don't use Windows at all, I don't think I will be doing these changes myself to get this working.
But feel free to make a pull request and I'll be glad to review, test and pull a Windows implementation.

**Reply by GhostNaN** [reply]
PS. If you do bother to make a Windows implementation, please make it a separate file to avoid spaghetti code, thanks.
